zebra/zebra_srv6_vty.c

```diff
@@ -345,6 +410,37 @@ DEFPY (locator_prefix,
 	return CMD_SUCCESS;
 }
 
+DEFUN_NOSH (srv6_encap,
+            srv6_encap_cmd,
+            "encapsulation",
+            "Segment Routing SRv6 encapsulation\n")
+{
+	vty->node = SRV6_ENCAP_NODE;
+	return CMD_SUCCESS;
+}
+
+DEFPY (srv6_src_addr,
+       srv6_src_addr_cmd,
+       "source-address X:X::X:X$encap_src_addr",
+       "Segment Routing SRv6 source address\n"
+       "Specify source address for SRv6 encapsulation\n")
+{
+	zebra_srv6_encap_src_addr_set(&encap_src_addr);
+	dplane_sr_tunsrc_set(&encap_src_addr, NS_DEFAULT);
+	return CMD_SUCCESS;
+}
+
+DEFPY (no_srv6_src_addr,
+       no_srv6_src_addr_cmd,
+       "no source-address",
```

Comment:
We should allow optionally specifying `X:X::X:X` source address.

Reply:
Changed the command to `no source-address [X:X::X:X$encap_src_addr]`. Thanks.